src/processor/mapper/map_extend.cpp

```diff
@@ -76,28 +87,45 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapLogicalExtendToPhysical(
         outputVectorsPos.emplace_back(outSchema->getExpressionPos(*expression));
     }
     auto& relsStore = storageManager.getRelsStore();
+    if (!rel->isDirected()) {
+        // Currently only implemented for single labelled rel and boundNode.
+        assert(!rel->isMultiLabeled() && !boundNode->isMultiLabeled());
+        std::unordered_map<table_id_t, std::unique_ptr<RelTableCollection>>
```

Comment:
Let's rename all `RelTableCollection` to `RelTableDataCollection` both here and in the GenericScanRelTable operator.